docs/handling-delete-in-ktb.rst

```diff
@@ -0,0 +1,63 @@
+Handling delete in the Kafka Topology Builder
+*******************************
+
+As the reader might already be aware, the Kafka Topology Builder is capable of handling deletes for you.
+This is like this in purpose as keeping the final state of the cluster consistent with the declarative picture you wrote in the descriptors is a core feature of the tool.
+
+However in some situations having delete enabled might not be what you are looking for.
+For this reason the tool allows you to control it in full granularity.
+
+Delete flag in the CLI
+-----------
+
+The CLI provide you currently with a global DELETE flag, this would allow deletion of all resources controlled by the Kafka Topology builder.
+The user can set this flag (*--allowDelete*) from the CLI as described.
+
+.. code-block:: bash
+
+    $> kafka-topology-builder.sh  --help
+        usage: cli
+          --allowDelete          Permits delete operations for topics and
+                                 configs. (deprecated, to be removed)
+          --brokers <arg>        The Apache Kafka server(s) to connect to.
+          --clientConfig <arg>   The AdminClient configuration file.
+          --dryRun               Print the execution plan without altering
+                                  anything.
+          --help                 Prints usage information.
+          --quiet                Print minimum status update
+          --topology <arg>       Topology config file.
+          --version              Prints useful version information.
+
+*NOTE*: This is a global flag, allowing/deny all delete operations.
+
+By default the value for this variable is false, so the Kafka Topology Builder will **not** delete any resource.
```

Comment:
I thought this was the default behaviour, but upon checking...not so. Ref TopologyBuilderConfig:

```
  public boolean allowDelete() {
    return Boolean.parseBoolean(cliParams.getOrDefault(BuilderCLI.ALLOW_DELETE_OPTION, "true"));
  }
```

Reply:
Should fix the default not the docs though.

Reply:
Not like this, the mistake is in using this default. but check the BuilderCLi class there true only if exist the flag.

Reply:
Ok, I see now. So should not use default here then?

Reply:
I like the default because I have as well build https://github.com/kafka-ops/kafka-gitops-api where I use the generated JAR as a lib. This is a "pet" project of mine but something I like to work as well in the future.

I will add false as default as It should have been done long enough

Reply:
👍